requirements_test.txt

```diff
@@ -31,4 +31,5 @@ trame>=2.5.2,<3.6.6
 trame-vtk>=2.5.8,<2.8.11
 trame-vuetify>=2.3.1,<2.7.2
 trimesh<4.5.0
+types-docutils
```

Comment:
We pin all the versions in this file.

Reply:
I'm also puzzled why we need this one as it isn't in pre-commit.  Is it due to not excluding the same files?  

Also do we know whether we need `mypy-extensions`?  Unless we know it is likely safer to keep this as like-for-like as possible.
```yaml
# list of dependencies in pre-commit mypy
            "mypy-extensions==1.0.0",
            "toml==0.10.2",
            "numpy>=2.0",
            "npt-promote==0.1",
            "matplotlib<3.9.1",
```

Reply:
This might have something to do with how `mypy` follows imports. With this workflow we run `mypy pyvista` and let `mypy` find everything it needs. But with pre-commit `mypy` is run separately on a list of files, not a directory. (Note, pre-commit will also run on files from `doc` and `tests` as well, which are technically excluded/not covered by the  new workflow).

I think including `types-docutils` is necessary here since there are a couple of errors raised otherwise.

The ignores needed for this to work are included with #6739, which is passing pre-commit CI. So, if we want this to be like-for-like as much as possible, we can create a separate PR first where we add `types-docutils` to the pre-commit dependencies along with the few ignores needed.

This example also highlights why this new workflow is needed: pre-commit does not appear to be catching errors like this.

Reply:
Figured it out. `types-docutils` this is only needed for `pyvista/ext` which should otherwise be excluded. See https://github.com/pyvista/pyvista/actions/runs/11533261412/job/32105976694?pr=6738#step:8:1

So we can remove this dependency.